pontoon/contributors/utils.py

```diff
@@ -420,20 +465,81 @@ def get_project_locale_contribution_counts(contributions_qs):
                     "name": item["locale_name"],
                     "code": item["locale_code"],
                 },
-                "actions": [action],
+                "action_dates": [
+                    {
+                        "action": action,
+                        "date": date,
+                    }
+                ],
                 "count": count,
             }
-
     return counts
 
 
-def get_contribution_timeline_data(user, contribution_type=None, day=None):
+def get_monthly_timeline_titles(contributions):
+    """
+    Regenerate titles and counts for each month and contribution_type
+    """
+    for date, types in contributions.items():
+        for contribution_type, entries in types.items():
+            c_count = sum(entry["count"] for entry in entries)
+            p_count = len(entries)
+
+            if contribution_type == "user-translations":
+                title = f"Submitted { intcomma(c_count) } translation{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "user-reviews":
+                title = f"Reviewed { intcomma(c_count) } suggestion{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "peer-reviews":
+                title = f"Received review for { intcomma(c_count) } suggestion{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+
+            contributions[date][contribution_type] = {
+                "title": title,
+                "entries": entries,
+            }
+
+    return contributions
+
+
+def combine_actions(actions_list):
+    """
+    Combine print statements for identical actions in the same project and month
+    """
+    combined = defaultdict(int)
+    total_count = 0
+    for action in actions_list:
+        parts = action.split()
+        count = int(parts[0])
+        action_type = " ".join(parts[1:])
+        if "translation" in action_type:
+            action_type = "translation"
+        combined[action_type] += count
+        total_count += count
+
+    combined_actions = []
+    for action_type, count in combined.items():
+        if "translation" in action_type:
+            action = f"{count} translation{pluralize(count)}"
+        else:
+            action = f"{count} {action_type}"
+        combined_actions.append(action)
+
+    return combined_actions, total_count
+
+
+def get_contribution_timeline_data(
+    user, year_shown=False, contribution_type=None, day=None
+):
     """
     Get data required to render the Contribution timeline on the Profile page
     """
     end = timezone.now()
-    start = end - relativedelta(months=1)
-    timeline_title = "Contribution activity in the last month"
+
+    if year_shown:
+        start = end - relativedelta(years=1, day=1)
+        timeline_title = "Contribution activity in the last year"
```

Comment:
How about we just hardcode this to "Contribution activity" in the template?

Then we'd need to always show month + year in the timeline - even for this month's data - which would result in a more consistent UX and simpler code with less IF-ELSE blocks.